components/validator/guardian.go

```diff
@@ -223,20 +214,20 @@ func (g *Guardian) outputRootAtBlock(ctx context.Context, l2BlockNumber uint64)
 	return output.OutputRoot, nil
 }
 
-func (g *Guardian) getL2OutputIndexAfter(ctx context.Context, l2BlockNumber *big.Int) (*big.Int, error) {
-	cCtx, cCancel := context.WithTimeout(ctx, g.cfg.NetworkTimeout)
+func (g *Guardian) getL2OutputIndexAfter(l2BlockNumber *big.Int) (*big.Int, error) {
+	cCtx, cCancel := context.WithTimeout(g.ctx, g.cfg.NetworkTimeout)
 	defer cCancel()
 	return g.l2ooContract.GetL2OutputIndexAfter(utils.NewSimpleCallOpts(cCtx), l2BlockNumber)
 }
 
-func (g *Guardian) isOutputFinalized(ctx context.Context, outputIndex *big.Int) (bool, error) {
+func (g *Guardian) IsOutputFinalized(ctx context.Context, outputIndex *big.Int) (bool, error) {
```

Comment:
It looks like u've removed all context arguments from the Guardian receiver, can we do something similar here?